format/amf3/Tools.hx

```diff
@@ -56,9 +58,15 @@ class Tools {
 			case cast Array:
 				var o : Array<Dynamic> = o;
 				var a = new Array();
-				for(v in o)
-					a.push(encode(v));
+				for(k in Reflect.fields(o))
```

Comment:
Why not using normal Array iteration here? Using Reflect.fields will break on most platforms.